src/vm/genericdict.h

```diff
@@ -92,6 +92,10 @@ typedef DPTR(DictionaryEntryLayout) PTR_DictionaryEntryLayout;
 class DictionaryLayout;
 typedef DPTR(DictionaryLayout) PTR_DictionaryLayout;
 
+// Number of slots to initially allocate in a generic method dictionary layout.
+// Also the number of additional slots to add to an existing dictionary layout when expanding its size
+#define NUM_DICTIONARY_SLOTS 8
```

Comment:
Why is 8 a good number? We used to run with 4 just fine.

Also, should this be a very small number like 1 in debug builds to stress this?

Reply:
I was under the impression that 4 wasn't enough, and it was just a number from the old fragile ngen days. I can make it 4, but with 8 we will get fewer expansions, so it might be better overall for perf. 

> Also, should this be a very small number like 1 in debug builds to stress this?

That's a good idea, I was actually using 1 during testing, but I think it would make more sense to put that under a complus stress mode instead of debug mode

Reply:
> it might be better overall for perf.

Depends on what is the performance metric you are looking at. Larger number will lead to more memory consumption that is not good for perf.

Since you are adding ability to expand, we should have have smaller default than what we used to have. The existing number was chosen conservatively because of we did not have ability to expand.

>  make more sense to put that under a complus stress mode instead of debug mode

We do have number of places that have different defaults for debug vs. release. The dedicated stress modes are expansive because of we need to have dedicated runs to execute them. It makes sense to create stress modes for these only if the smaller default for debug would be too slow.

Reply:
Thanks for feedback Jan. In that case, i'll make it either 4 or 2. I'm measuring some numbers for both cases.
In terms of memory usage, here are the numbers for the msix catalog WPF app:
- original: 54.1 MB
- With 4 slots, and expansions add 4 slots at a time: 54.5 MB
- With 2 slots, and expansions add 2 slots at a time: 54.15 MB.

There isn't that big of a difference (0.09% to 0.7% increase), so the real factor would be the startup perf numbers here.

Note that this number only affects generic methods, and generic types that start running out of slots. There's another heuristic that dictates how many initial slots to create for generic types.

> We do have number of places that have different defaults for debug vs. release

Sounds good. I'll make it 1 for debug builds.